core/domain/suggestion_registry.py

```diff
@@ -0,0 +1,329 @@
+        self.status = status
+        self.author_id = author_id
+        self.assigned_reviewer_id = assigned_reviewer_id
+        self.final_reviewer_id = final_reviewer_id
+        self.change_cmd = change_cmd
+        self.score_category = score_category
+
+    def validate(self):
+        """Validates a suggestion object of type
+        SUGGESTION_TYPE_EDIT_STATE_CONTENT.
+
+        Raises:
+            ValidationError: One or more attributes of the
+                SuggestionEditStateContent object are invalid.
+        """
+        super(SuggestionEditStateContent, self).validate()
+
+        if self.get_score_type() != suggestion_models.SCORE_TYPE_CONTENT:
+            raise utils.ValidationError(
+                'Expected the first part of score_category to be %s '
+                ', received %s' % (
+                    suggestion_models.SCORE_TYPE_CONTENT,
+                    self.get_score_type()))
+
+        if self.get_score_sub_type() not in constants.ALL_CATEGORIES:
+            raise utils.ValidationError(
+                'Expected the second part of score_category to be a valid'
+                ' category, received %s' % self.get_score_sub_type())
+
+        if 'cmd' not in self.change_cmd:
+            raise utils.ValidationError(
+                'Expected change_cmd to contain a cmd key')
+
+        if self.change_cmd['cmd'] != exp_domain.CMD_EDIT_STATE_PROPERTY:
+            raise utils.ValidationError(
+                'Expected cmd to be %s, received %s' % (
+                    exp_domain.CMD_EDIT_STATE_PROPERTY, self.change_cmd['cmd']))
+
+        if 'property_name' not in self.change_cmd:
+            raise utils.ValidationError(
+                'Expected change_cmd to contain a property_name key')
+
+        if (
+                self.change_cmd['property_name'] !=
+                exp_domain.STATE_PROPERTY_CONTENT):
+            raise utils.ValidationError(
+                'Expected property_name to be %s, received %s' % (
+                    exp_domain.STATE_PROPERTY_CONTENT,
+                    self.change_cmd['property_name']))
+
+        if 'state_name' not in self.change_cmd:
+            raise utils.ValidationError(
+                'Expected change_cmd to contain a state_name key')
+
+        states = exp_services.get_exploration_by_id(self.target_id).states
+        if self.change_cmd['state_name'] not in states:
+            raise utils.ValidationError(
+                'Expected %s to be a valid state name' %
+                self.change_cmd['state_name'])
+
+    def update_change_cmd_before_accept(self):
+        """Before accepting the suggestion, modifications need to be done to the
+        change_cmd.
+        """
+        exploration = exp_services.get_exploration_by_id(self.target_id)
+        old_content = exploration.states[self.change_cmd['state_name']].content
+
+        self.change_cmd['old_content'] = old_content
```

Comment:
For this type of suggestion we don't need the check. What this function does is it stores the value of the content just before accepting as the old_content (so that if the change is reverted it goes back to the correct content). 

When we create the translations sub class, in its validate() we will have the check for old content being same as current content. The validate() is called just before accepting.

Reply:
Doesn't that apply to exploration content as well? If the old_content has changed significantly, I don't think that suggestion will be valid. But we can leave that for the reviewers to decide.

Reply:
Yeah. The reviewer can reject if it is out of context since when the suggestion was made. This is not something (I think) we can detect in code. If the content is different but the state is still present, theoretically, the suggestion can still be accepted.